Tests/NIOTransportServicesTests/NIOTSEndToEndTests.swift

```diff
@@ -557,5 +557,49 @@ class NIOTSEndToEndTests: XCTestCase {
                 print(error)
         }
     }
+    
+    func testViabilityUpdate() throws {
+        final class ViabilityHandler: ChannelInboundHandler {
+            typealias InboundIn = ByteBuffer
+            
+            private let testCompletePromise: EventLoopPromise<Bool>
+            let listenerChannel: Channel
+            
+            init(testCompletePromise: EventLoopPromise<Bool>, listenerChannel: Channel) {
+                self.testCompletePromise = testCompletePromise
+                self.listenerChannel = listenerChannel
+            }
+
+            func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+                if event is NIOTSNetworkEvents.ViabilityUpdate {
+                    let update = event as! NIOTSNetworkEvents.ViabilityUpdate
```

Comment:
You can just use `if let as` to avoid the `as!`:

```suggestion
                if let update = event as? NIOTSNetworkEvents.ViabilityUpdate {
```